### PR TITLE
[dma/dv] Fixes and updates to testplan

### PR DESCRIPTION
**Pull request by andreaskurth**
Please see commit messages for details.

### CODE REVIEW

File: hw/ip/dma/data/dma_testplan.hjson [hunk 1]
```diff
@@ -245,7 +245,7 @@
               - Check if PLIC interrupt is asserted when memory_buffer_auto_increment is enabled
               '''
         stage: V2
-        tests: ["dma_handshake_mem_buffer_overflow"]
+        tests: ["dma_handshake_stress"]
```
Comment (andreaskurth) [inline hw/ip/dma/data/dma_testplan.hjson:248]: @alees24: Could you please confirm that the `dma_handshake_stress` test includes random configurations that exercise the memory buffer threshold and limit behavior?

File: hw/ip/dma/data/dma_testplan.hjson [hunk 2]
```diff
@@ -210,6 +210,14 @@
         stage: V2
         tests: ["dma_memory_stress"]
       }
+      {
+        name: dma_generic_stress
+        desc: '''
+              Stress test in which random DMA operations are performed with 'memory-to-memory' and 'hardware handshaking' mode.
```
Comment (andreaskurth) [inline hw/ip/dma/data/dma_testplan.hjson:216]: @alees24: You may be able to provide a more detailed description of this test point?

File: hw/ip/dma/data/dma_testplan.hjson [hunk 3]
```diff
@@ -316,7 +316,7 @@
               - Check if DMA generates an alert
               '''
         stage: V2S
-        tests: ["dma_illegal_addr_range"]
+        tests: ["dma_generic_stress", "dma_handshake_stress"]
```
Comment (andreaskurth) [inline hw/ip/dma/data/dma_testplan.hjson:319]: @alees24: Could you please confirm that the `dma_generic_stress` and `dma_handshake_stress` tests include random configurations that exercise illegal source or destination addresses?

File: hw/ip/dma/data/dma_testplan.hjson [hunk 4]
```diff
@@ -295,7 +295,7 @@
               - Check if DMA generates an alert
               '''
         stage: V2S
-        tests: ["dma_illegal_transfer"]
+        tests: []
```
Comment (alees24) [inline hw/ip/dma/data/dma_testplan.hjson:298]: My understanding is that it is *not* illegal to transfer between CTN and SYS, and we certainly don't prohibit this in the DMA controller presently, nor can I find any mention of such a requirement in the original specification document.
Maybe we should just remove this test point?

File: hw/ip/dma/data/dma_testplan.hjson (hunk 3 above)
Comment (alees24) [reply hw/ip/dma/data/dma_testplan.hjson:319]: Yes, they do, but 'dma_mem_enabled' is better at it; another case of a more directed sequence to increase probability.

**Review by alees24 — CHANGES_REQUESTED** [review]
Thanks for this. There are probably a couple of small refinements - see comments - that should probably be made whilst the testplan is receiving attention, please.

File: hw/ip/dma/data/dma_testplan.hjson (hunk 4 above)
Comment (andreaskurth) [reply hw/ip/dma/data/dma_testplan.hjson:298]: I agree that we [currently don't specify restrictions for transfers between CTN and SYS](https://github.com/lowRISC/opentitan/blob/integrated_dev/hw/ip/dma/doc/theory_of_operation.md#security-considerations), and the way the `dma_illegal_transfer` testpoint is described, any direct transfers between CTN and SYS would be illegal. So I'm in favor of removing this testpoint without substitution.

@Razer6: Does this SGTY as well?

Comment (Razer6) [reply hw/ip/dma/data/dma_testplan.hjson:298]: Yes, there are no such restrictions. Removing that testpoint sounds good to me.

Comment (andreaskurth) [reply hw/ip/dma/data/dma_testplan.hjson]: Thx, removed.

File: hw/ip/dma/data/dma_testplan.hjson (hunk 1 above)
Comment (andreaskurth) [reply hw/ip/dma/data/dma_testplan.hjson]: Thx, added `dma_mem_limit`

**Comment by andreaskurth** [comment]
Thanks for your feedback, @alees24! I think I implemented the points you raised; could you PTAL again?

**Review by Razer6 — APPROVED** [review]

**Review by alees24 — APPROVED** [review]
Thanks for the updates.

**Pull request merged**